esmvaltool/cmorizers/obs/cmorize_obs_hadcrut5.py

```diff
@@ -0,0 +1,105 @@
+"""ESMValTool CMORizer for HadCRUT5 data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+    https://crudata.uea.ac.uk/cru/data/temperature
+
+Last access
+    20210113
+
+Download and processing instructions
+    Download the following files:
+        infilling
+            [Source]/HadCRUT.5.0.0.0.analysis.anomalies.ensemble_mean.nc
+        no-infilling
+            [Source]/HadCRUT.5.0.0.0.anomalies.ensemble_mean.nc
+        climatology
+            [Source]/absolute_v5.nc
+"""
+
+import copy
+import logging
+import os
+
+import iris
+import numpy as np
+from cf_units import Unit
+
+from . import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def _extract_variable(short_name, var, version, filename, cfg, in_dir,
+                      out_dir):
+    """Extract variable."""
+    # load data
+    filepath = os.path.join(in_dir, filename)
+    raw_var = var.get('raw', short_name)
+    cube = iris.load_cube(filepath, utils.var_name_constraint(raw_var))
+
+    # load climatology
+    filepath_clim = os.path.join(in_dir, cfg['climatology']['filename'])
+    raw_var = var.get('raw_clim', short_name)
+    clim_cube = iris.load_cube(filepath_clim,
+                               utils.var_name_constraint(raw_var))
+
+    # fix units
+    cmor_info = cfg['cmor_table'].get_variable(var['mip'], short_name)
+    for cub in [cube, clim_cube]:
+        if cub.units != cmor_info.units:
+            cub.convert_units(cmor_info.units)
+
+    # derive absolute temperatures
+    clim_data = clim_cube.data
+    clim_data = np.tile(clim_data, [cube.shape[0] // 12, 1, 1])
+    if cube.shape[0] % 12 != 0:
+        for i in range(cube.shape[0] % 12):
+            clim_data = np.vstack([clim_data, clim_data[i:i + 1]])
+
+    cube.data = cube.data + clim_data
+
+    # fix time units
+    cube.coord('time').convert_units(
+        Unit('days since 1950-1-1 00:00:00', calendar='gregorian'))
+
+    # Fix coordinates
+    utils._fix_dim_coordnames(cube)
+    cube_coord = cube.coord('longitude')
+    if cube_coord.points[0] < 0. and \
+            cube_coord.points[-1] < 181.:
+        cube_coord.points = \
+            cube_coord.points + 180.
```

Comment:
just use `utils.fix_coords(cube)` for this

Reply:
The problem with fix_coords is that it calls _fix_bounds on the time coordinate -> removing and reguessing bounds and thereby screwing up the monthly bounds via its linear interpolation ...

Reply:
aha good pointer! I'll put in a patch to `utilities.fix_coords()` via an optional argument to keep the bounds untouched :+1:

Reply:
note tha now we have a a set of new args to this function that allow the developer to turn off overwriting the bounds, so next time it's best to use that. It's fine here, no need to change the code :+1: